unit-tests/application_test.py

```diff
@@ -0,0 +1,11 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+# We do not unit test the applications as they are far too complex for it.
+# They would require too many fixtures and mocks to test. At that point our
+# unit test would have become a system test.
+#
+# TODO: This is probably an indication that the application classes contain
+#       too much stuff.
```

Comment:
Just to say I agree with this assessment - I probably should have added the new "compilation" step within its own new module.  Something for a future ticket for sure